dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/TrackedEntityCriteriaMapper.java

```diff
@@ -384,4 +369,17 @@ private void validateOrderParams( List<OrderParam> orderParams, Map<String, Trac
             }
         }
     }
+
+    private Set<String> getAssignedUsers( TrackerTrackedEntityCriteria criteria )
+    {
+        Set<String> assignedUsers = new HashSet<>();
+
+        if ( criteria.getAssignedUser() != null && !criteria.getAssignedUser().isEmpty() )
+        {
+            assignedUsers = TextUtils.splitToSet( criteria.getAssignedUser(), TextUtils.SEMICOLON ).stream()
+                .filter( CodeGenerator::isValidUid ).collect( Collectors.toSet() );
```

Comment:
I am just wondering why we remove invalid UIDs only for assigned users but not for OrgUnits and TrackedEntityInstances

Reply:
I was also wondering. I have no clue 🤷🏻 In this PR I just move code around. I took the code from the TrackedEntityInstanceCriteria. So this is how it was.

I think we should discuss this topic. I don't think its a good idea to remove invalid UIDs in any case. As a user I would expect all my parameters to be taken into account. If my request is not valid I expect an error.

Reply:
We will discuss this in todays design meeting 😄 Lets keep the behavior as it is for now. I would see this as a dedicated task/issue where we need to go through all our endpoints and ensure we return an error instead of filtering the UIDs out.